lib/discourse_chat/provider/slack/slack_message.rb

```diff
@@ -46,14 +62,35 @@ def text
           next "@#{user_name}"
         end
 
-        "[#{text}](#{link})"
+        if link.nil?
+          text
+        elsif link == text
+          "<#{link}>"
+        else
+          "[#{text}](#{link})"
+        end
       end
 
       # Add an extra * to each side for bold
-      text = text.gsub(/\*(.*?)\*/) do |match|
+      text = text.gsub(/\*.*\*/) do |match|
         "*#{match}*"
       end
 
+      # Add an extra ~ to each side for strikethrough
+      text = text.gsub(/~.*?~/) do |match|
+        "~#{match}~"
+      end
+
+      # Replace emoji - with _
+      text = text.gsub(/:[a-z0-9_-]+:/) do |match|
```

Comment:
Shorter?

```suggestion
      text = text.gsub(/:[\w-]+:/) do |match|
```

Reply:
Slack emojis must be all-lowercase, and I think `\w` would include uppercase characters